boms/integration-tests/src/test/java/com/google/cloud/MaximumLinkageErrorsTest.java

```diff
@@ -70,13 +76,24 @@ public void testMaximumLinkageErrors()
       ImmutableSet<ClassFile> currentReferences = currentProblems.get(problem);
       SetView<ClassFile> newReferences = Sets.difference(currentReferences, oldReferences);
       if (!newReferences.isEmpty()) {
-        StringBuilder message =
-            new StringBuilder("Newly introduced classes linking to " + problem + ":\n");
+        message.append("Newly introduced classes linking to " + problem + ":\n");
         for (ClassFile classFile : newReferences) {
           message.append("Link from " + classFile + "\n");
         }
         Assert.fail(message.toString());
       }
     }
   }
+
+  private String findLatestNonSnapshotVersion() throws MavenRepositoryException {
+    ImmutableList<String> versions =
+        RepositoryUtility.findVersions(newRepositorySystem(), "com.google.cloud", "libraries-bom");
```

Comment:
This method doesn't seem to promise returning the version in any particular order.

Reply:
Clarified that in the javadoc in `findVersions`. [Javadoc for ` VersionRangeResult.getVersions()`](https://maven.apache.org/resolver/apidocs/org/eclipse/aether/resolution/VersionRangeResult.html#getVersions()) ensures the order.